lib/ansible/modules/cloud/vagrant/vagrant_box.py

```diff
@@ -0,0 +1,290 @@
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: vagrant_box
+short_description: Manage Vagrant boxes
+version_added: "2.5"
+author: "Chafik Belhaoues (@africanzoe)"
+description:
+   - This module can add|remove|update a Vagrant box on a target host.
+options:
+    box:
+      description:
+        - The name of the box to be managed, it could be a path to a file, URL or shorthand name from Vagrant cloud.
+      aliases: ['path', 'url']
+      required: true
+      default: None
+    state:
+      description:
+        - Indicate the desired state of the box
+      choices: ['present', 'absent', 'updated']
+      default: present
+    force:
+      description:
+        - Specify wether to force the add or remove actions.
+      default: false
+    version:
+      description:
+        - Specify the version of the managed box.
+        - Please consider using an explicit version with the state "absent".
+      default: latest
+    provider:
+      description:
+        - Specify the provider for the box.
+      choices: ['virtualbox', 'libvirt', 'hyperv', 'vmware_desktop']
+      default: "virtualbox"
+    insecure:
+      description:
+        - Indicate that the SSL certificates will not be verified if the URL is an HTTPS URL.
+      default: false
+    trust_location:
+      description:
+        - Indicate wether to trust 'Location' header from HTTP redirects and use the same credentials for subsequent urls as for the initial one or not.
+      default: false
+    cacert:
+      description:
+        - The path on the target host to the FILE that contains CA certificate for SSL download.
+    capath:
+      description:
+        - The path on the target host to the DIRECTORY for the CA used to verify the peer.
+      default: false
+    cert:
+      description:
+        - The path on the target host to the a client certificate to use when downloading the box, if necessary.
+    name:
+      description:
+        - The name to give to the box if added from a file.
+        - When adding a box from a catalog, the name is included in the catalog entry and should not be specified.
+      default: the basename of the file
+    checksum:
+      description:
+        - The checksum for the box that is downloaded.
+    checksum_type:
+      description:
+        - The hash algorithm used for the checksum.
+      choices: ['md5', 'sha1', 'sha256']
+      default: sha256
+'''
+EXAMPLES = '''
+---
+- name: Add Debian 9 box
+  vagrant_box:
+    box: "debian/stretch64"
+    state: "present"
+    provider: "virtualbox"
+    version: "9.2.0"
+    force: false
+
+- name: Remove Ubuntu xenial64 box
+  vagrant_box:
+    box: "ubuntu/xenial64"
+    state: "absent"
+    provider: "virtualbox"
+    version: "20170914.2.0"
+
+- name: Update Centos 7 box
+  vagrant_box:
+    box: "centos/7"
+    state: "updated"
+    provider: "virtualbox"
+'''
+
+RETURN = '''
+changed:
```

Comment:
`changed` is a common return and must not be documented in a module. please remove it.